File: examples/java/kora-java-database-cassandra/src/main/java/io/koraframework/example/cassandra/RootService.java
package io.koraframework.example.cassandra;

import io.koraframework.common.annotation.Component;
import io.koraframework.common.annotation.Root;

@Root
@Component
public final class RootService {

    private final CassandraCrudSyncRepository crudSyncRepository;
    private final CassandraCrudAsyncRepository crudAsyncRepository;
    private final CassandraMapperResultSetRepository mapperResultSetRepository;
    private final CassandraMapperRowRepository mapperRowRepository;
    private final CassandraMapperRowColumnRepository mapperRowColumnRepository;
    private final CassandraMapperParameterRepository mapperParameterRepository;
    private final CassandraUdtRepository cassandraUdtRepository;

    public RootService(CassandraCrudSyncRepository crudSyncRepository,
                       CassandraCrudAsyncRepository crudAsyncRepository,
                       CassandraMapperResultSetRepository mapperResultSetRepository,
                       CassandraMapperRowRepository mapperRowRepository,
                       CassandraMapperRowColumnRepository mapperRowColumnRepository,
                       CassandraMapperParameterRepository mapperParameterRepository,
                       CassandraUdtRepository cassandraUdtRepository) {
        this.crudSyncRepository = crudSyncRepository;
        this.crudAsyncRepository = crudAsyncRepository;
        this.mapperResultSetRepository = mapperResultSetRepository;
        this.mapperRowRepository = mapperRowRepository;
        this.mapperRowColumnRepository = mapperRowColumnRepository;
        this.mapperParameterRepository = mapperParameterRepository;
        this.cassandraUdtRepository = cassandraUdtRepository;
    }
}

File: examples/kotlin/kora-kotlin-database-cassandra/src/main/kotlin/io/koraframework/kotlin/example/cassandra/RootService.kt
package io.koraframework.kotlin.example.cassandra

import com.datastax.oss.driver.api.core.cql.ResultSet
import com.datastax.oss.driver.api.core.cql.Row
import com.datastax.oss.driver.api.core.data.GettableByName
import com.datastax.oss.driver.api.core.data.SettableByName
import io.koraframework.common.annotation.Component
import io.koraframework.common.annotation.Mapping
import io.koraframework.common.annotation.Root
import io.koraframework.database.cassandra.CassandraRepository
import io.koraframework.database.cassandra.annotation.EntityCassandra
import io.koraframework.database.cassandra.annotation.UDT
import io.koraframework.database.cassandra.mapper.parameter.CassandraParameterColumnMapper
import io.koraframework.database.cassandra.mapper.result.CassandraResultSetMapper
import io.koraframework.database.cassandra.mapper.result.CassandraRowColumnMapper
import io.koraframework.database.cassandra.mapper.result.CassandraRowMapper
import io.koraframework.database.common.annotation.Batch
import io.koraframework.database.common.annotation.Column
import io.koraframework.database.common.annotation.Query
import io.koraframework.database.common.annotation.Repository

@Root
@Component
class RootService(
    private val crudSyncRepository: CassandraCrudSyncRepository,
    private val cassandraUdtRepository: CassandraUdtRepository,
    private val mapperRowColumnRepository: CassandraMapperRowColumnRepository,
    private val mapperParameterRepository: CassandraMapperParameterRepository,
    private val mapperRowRepository: CassandraMapperRowRepository,
    private val mapperResultSetRepository: CassandraMapperResultSetRepository
)
